azuredevops/internal/service/git/data_git_repository_file.go

```diff
@@ -0,0 +1,124 @@
+package git
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/microsoft/azure-devops-go-api/azuredevops/v7/git"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/client"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/utils"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/utils/converter"
+)
+
+func DataGitRepositoryFile() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceGitRepositoryFileRead,
+		Timeouts: &schema.ResourceTimeout{
+			Read: schema.DefaultTimeout(5 * time.Minute),
+		},
+		Schema: map[string]*schema.Schema{
+			"repository_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  "The repository ID",
+				ValidateFunc: validation.IsUUID,
+			},
+			"file": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The file path",
+			},
+			"branch": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				Description:   "The branch name, no default",
+				ConflictsWith: []string{"tag"},
```

Comment:
At least one of `tag` or `branch` must be set can set here instead of checking in the code.
```suggestion
				ConflictsWith: []string{"tag"},
				AtLeastOneOf:  []string{"branch", "tag"},
```

Reply:
Use ExactlyOneOf and added ValidateFunc: validation.StringIsNotWhiteSpace to both branch and tag